File: frontend/src/types/artwork.ts
export type Difficulty = "beginner" | "intermediate" | "advanced";

export interface Stage {
  number: number;
  title: string;
  description: string;
  duration_min: number;
  image?: string;   // static path for thumbnail, e.g. /artworks/mujer-azul-divisions/etapa-01.png
  objective?: string;
  colors?: string[];
  materials?: string[];
  brushes?: string[];
}

export interface Artwork {
  id: string;
  title: string;
  artist: string;
  year: number;
  difficulty: Difficulty;
  stages: Stage[];
  duration_min: number;   // total
  color: string;          // fallback background color
  image: string;          // URL de la obra (Wikimedia Commons)
  tags: string[];
}

File: rpi5/backend/app/models/artwork.py
from sqlalchemy import String, Integer, JSON
from sqlalchemy.orm import Mapped, mapped_column
from app.db.base import Base


class Artwork(Base):
    __tablename__ = "artworks"

    id:           Mapped[str] = mapped_column(String, primary_key=True)
    title:        Mapped[str] = mapped_column(String, nullable=False)
    artist:       Mapped[str] = mapped_column(String, nullable=False)
    year:         Mapped[int] = mapped_column(Integer, nullable=False)
    difficulty:   Mapped[str] = mapped_column(String, nullable=False)
    duration_min: Mapped[int] = mapped_column(Integer, nullable=False)
    color:        Mapped[str] = mapped_column(String, nullable=False)
    image:        Mapped[str] = mapped_column(String, nullable=False)
    tags:         Mapped[list] = mapped_column(JSON, default=list)
    stages:       Mapped[list] = mapped_column(JSON, default=list)
